rr_common/src/follower/follower.cpp

```diff
@@ -0,0 +1,100 @@
+#include "follower.h"
+using namespace std;
+
+void mapCallback(const sensor_msgs::PointCloud2ConstPtr& map) {
+    pcl::PCLPointCloud2 pcl_pc2;
+    pcl_conversions::toPCL(*map, pcl_pc2);
+    pcl::PointCloud<pcl::PointXYZ>::Ptr cloud(new pcl::PointCloud <pcl::PointXYZ>);
+    pcl::fromPCLPointCloud2(pcl_pc2, *cloud);
+    rr_platform::speedPtr speedMSG(new rr_platform::speed);
+    rr_platform::steeringPtr steerMSG(new rr_platform::steering);
+    if (cloud->empty()) {
+        ROS_WARN("environment map pointcloud is empty");
+        speedMSG->speed = 0;
+        steerMSG->angle = 0;
+        steerMSG->header.stamp = ros::Time::now();
```

Comment:
I am not sure how this timestamp is used, but you might want to check this makes semantic sense